packages/analytics-core/src/dma.js

```diff
@@ -0,0 +1,57 @@
+import AvAnalyticsPlugin from './plugin';
+
+const requiredFields = [
+  'tradingPartnerId',
+  'customerId',
+  'category',
+  'applicationId',
+];
+
+export default class AvDmaAnalytics extends AvAnalyticsPlugin {
+  constructor(AvLogMessages, enabled) {
+    super(enabled);
+    this.AvLogMessages = AvLogMessages;
+  }
+
+  trackEvent(properties) {
+    if (!properties) return {};
+
+    const logItems = {};
+
+    if (properties.ApplicationId) {
+      properties.applicationId = properties.ApplicationId;
+      delete properties.ApplicationId;
+    }
+
+    if (properties.Category) {
+      properties.category = properties.Category;
+      delete properties.Category;
+    }
+
+    if (!properties.tradingPartnerId || properties.TradingPartnerId) {
```

Comment:
This looks odd, just want to confirm:
if `properties.TradingPartnerId` is `'Something'`, it will set `properties.tradingPartnerId` to `'NA'` and delete `properties.TradingPartnerId`;

Reply:
Yeah I need to actually change this. The idea was that it was uppercasing the values that came in after parsing the different attributes. The "defaults" for tradingPartnerId and customerId should be set to "NA" for the log to work. 

However Rob and I talked about abstracting this to another microservice that will populate that rather than require a developer to add this in the UI which is the correct way to do it.